.github/workflows/hrm-ci.yml

```diff
@@ -4,47 +4,59 @@ on: [push]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
-    strategy:
-      matrix:
-        node-version: [14.x]
+    services:
+      localstack:
+        image: localstack/localstack:latest
+        env:
+          SERVICES: cloudformation,iam,sts,ssm,s3,cloudwatch,cloudwatch-logs,lambda,dynamodb,apigateway
+          DEFAULT_REGION: eu-west-1
+          AWS_ACCESS_KEY_ID: localkey
+          AWS_SECRET_ACCESS_KEY: localsecret
+        ports:
+          - 4566:4566
+          - 4571:4571
 
     steps:
-    - name: Checkout Branch
-      uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - name: Install Packages
-      working-directory: hrm-service
-      run: npm install
-    - name: Run Lint
-      working-directory: hrm-service
-      run: npm run lint --if-present
-    - name: Run Unit Tests
-      working-directory: hrm-service
-      run: npm run test:unit
-    - name: Setup database tables
-      uses: m4nu56/postgresql-action@1.0
-      with:
-        # Version of PostgreSQL to use
-        postgresql version: 12.7
-        # POSTGRES_DB - name for the default database that is created
-        postgresql db: hrmdb
-        # POSTGRES_USER - create the specified user with superuser power
-        postgresql user: hrm
-        # POSTGRES_PASSWORD - superuser password
-        postgresql password: postgres
-        # POSTGRES_INIT_SCRIPTS - directory containing DB init scripts
-        postgresql init scripts: hrm-service/service-tests/db-init
-    - name: Run Service Tests
-      working-directory: hrm-service
-      run: npm run test:service
-      env:
-        CI: true
-        TWILIO_ACCOUNT_SID: ACxxx
-        TWILIO_AUTH_TOKEN: xxxxxx
-        RDS_PASSWORD: postgres
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.x
+
+      - name: Install Packages
+        run: npm install
+
+      - name: Run Lint
+        run: npm run lint --if-present
+
+      - name: Run Unit Tests
+        run: npm run test:unit
+
```

Comment:
This is no longer relevant. I mocked SSM to eliminate the need for localstack in service tests, but if we wanted to do slightly more than even blackbox testing in github actions using localstack, it is possible. And in that case, I would suggest those tests run on PR, not push.

Reply:
Agree, we definitely want to run E2E / multi component tests on PR rather than push, that's what we do with our UI driven E2E tests in Flex.

If any E2E suites get very slow we can always consider only running them on release, but I'd prefer to avoid letting any particular test suite get that slow & unwieldy in the first place

I find black box single component tests to be more of a judgement call, if they're quick enough, run them on push, otherwise, run them on PR. Up until now they've been pretty fast to run so leaving them to run on push seemed fine. Perhaps as we add more for different components this will change